### ISSUE TITLE
Unable to use Money.Ecto.Type in a phoenix app

### ISSUE THREAD

**Issue by vraravam** [kipcole9/money#4]
I'm using the latest versions of the ex_money and ex_cldr packages.
```
      {:ex_cldr, "0.0.13"},
      {:ex_money, "~> 0.0.9"},
```
But, its not letting me use the `Money.Ecto.Type` as the type of the field.

<img width="960" alt="screen shot 2016-12-11 at 2 14 38 pm" src="https://cloud.githubusercontent.com/assets/69629/21079100/a7c50e4e-bfac-11e6-9f90-16ed88cf5405.png">

**Comment by kipcole9** [comment]
Looks like I didn't include documentation for the migration.  In your migration, the structure is:
```
defmodule MoneyTest.Repo.Migrations.CreateThing do
  use Ecto.Migration

  def change do
    create table(:things) do
      add :amount, :money_with_currency
      timestamps()
    end
  end
end
```
Notice that you use the type as defined in the database, i.e. `:money_with_currency`. Then in your schema;
```
defmodule MoneyTest.Repo.Migrations.CreateThing do
  use Ecto.Migration

  def change do
    create table(:things) do
      add :amount, :money_with_currency
      timestamps()
    end

  end
end
```

**Comment by kipcole9** [comment]
I suspect also the error is caused by my conditional compilation issues.  A forced recompile will likely help with this issue too:  `mix deps.compile ex_money --force`

**Comment by vraravam** [comment]
Thanks - the forced recompile did some bit of the trick. btw, I was using the migration and the model the same way as you described.

Now, the next issue is:
```
** (Postgrex.Error) ERROR (syntax_error): type modifier is not allowed for type "money_with_currency"
```
In my migration, I have the following:
```
add :amount, :money_with_currency, precision: 10, scale: 2
```
Is there any easy way to correct it? Or do you suggest that I let the Postgres Type definition work its magic?

**Comment by kipcole9** [comment]
If you want to change the definition of the precision and scale then I think the best way is to just change the generated migration.  As the message says, there's not way to customise the precision and scale (although i will look into that to see if the api supports the idea.

**Comment by vraravam** [comment]
The precision that you have specified in the migration is good enough for me to override - since that is (strictly speaking) not in the package code, and since its generated, i assumed as much that the author using your lib is free to change those bits. Thanks for the quick responses btw - and a great library.

**Comment by kipcole9** [comment]
Yes, you're right - the migration lives in your code, not in the library so you're free to change it any way you like.  I picked (20,8) since I think that caters for a wide range of currencies (Rupee, Rupiah, Dong as more obvious examples) and still preserve precision when doing math and currency conversions.  Thanks for the encouragement too, its a good push to iron out the early kinks.

**Comment by vraravam** [comment]
np - I appreciate all your help.

btw, when i was working for a financial services client in the US, the law
(at that time used to enforce a minimum of 6 digits of precision beyond the
decimal separator for data that got stored in any db. So, any number >6
would satisfy those kind of situations as well. Good choice.

**Comment by kipcole9** [comment]
Updated `ex_money` to version 0.0.11 which appears to fix this issue.

**Comment by vraravam** [comment]
Yes - this seems to be fixed with the upgrade to `ex_money v0.0.11`

**Comment by kipcole9** [comment]
Thanks for the confirmation.   I'm at a stable point in development on this library now - happy to take suggestions for further development if you see gaps that would be valuable to you.
